javascript/ql/src/semmle/javascript/security/dataflow/CleartextLogging.qll

```diff
@@ -64,6 +64,14 @@ module CleartextLogging {
         trg.(DataFlow::SourceNode).flowsTo(write.getBase())
       )
     }
+
+    override predicate isBarrier(DataFlow::Node src, DataFlow::Node trg) {
+      // property reads does not propagate  ...
+      trg instanceof DataFlow::PropRead and
+      // ... unless the predecessor is a source by itself
+      not src instanceof Source
```

Comment:
It seems counterproductive to add additional flow steps into objects, and then filter them out with a barrier.

Have you tried simply deleting the extra flow steps above? I think the improved flow through properties should handle the test case you added.